arkouda/categorical.py

```diff
@@ -529,3 +534,197 @@ def concatenate(self, others : List[Categorical], ordered : bool=True) -> Catego
+            If the Categorical object has no underlying register-able components.
+
+        See also
+        --------
+        unregister, attach, unregister_categorical_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        components = self._get_vars_with_register()
+        if components:
+            for k, v in components:
+                self.logger.debug(f"Registering attribute {k} with name {user_defined_name}_{k}")
+                v.register(f"{user_defined_name}.{k}")  # This builds out the registration name dynamically
+        else:
+            raise RegistrationError("Categorical object did not have any underlying register-able components")
+
+        self.name = user_defined_name
+        return self
+
+    def unregister(self) -> None:
+        """
+        Unregister this Categorical object in the arkouda server which was previously
+        registered using register() and/or attached to using attach()
+
+        Raises
+        ------
+        RegistrationError
+            If the object is already unregistered or if there is a server error
+            when attempting to unregister
+
+        See also
+        --------
+        register, attach, unregister_categorical_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        if not self.name:
+            raise RegistrationError("This item does not have a name and does not appear to be registered.")
+
+        components = self._get_vars_with_register()
+        if components:
+            for k, v in components:
+                self.logger.debug(f"Un-registering Categorical attribute {k} with name {self.name}.{k}")
+                v.unregister()
+        self.name = None  # Clear our internal Categorical object name
+
+    def is_registered(self) -> np.bool_:
+        """
+         Return True iff the object is contained in the registry
+
+        Returns
+        -------
+        numpy.bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mis-match of registered pieces
+
+        See Also
+        --------
+        register, attach, unregister, unregister_categorical_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        reg:bool
+        components = self._get_vars_with_register()
+        if components:
+            registered = [k for k, v in components if v.is_registered()]
+            if registered and len(registered) == len(components):  # everything is registered
+                reg = True
```

Comment:
no change requested but I was wondering if something like this would work here
```python
registered = [v.is_registered() for k, v in components.items()]
if registered and np.all(registered):
    reg = True
```
It doesn't change the functionality but using `np.all` seems more straightforward and pythonic to me than the length comparison.

Reply:
I see what you are saying and I think you are correct.  However, this piece of code is being pitched.

Reply:
👍